examples/plots/Four_Panel_Map.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2008-2016 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+Four Panel Map
+===============
+
+By reading model output data from a netCDF file, we can create a four panel plot showing:
+
+* 300 hPa heights and winds
+* 500 hPa heights and absolute vorticity
+* Surface temperatures
+* Precipitable water
+"""
+
+###########################################
+
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import matplotlib.gridspec as gridspec
+import matplotlib.pyplot as plt
+import netCDF4
+import numpy as np
+import scipy.ndimage as ndimage
+
+from metpy.cbook import get_test_data
+from metpy.units import units
+
+###########################################
+
+
+# Function used to create the map subplots
+def plot_background(ax):
+    ax.set_extent([235., 290., 20., 55.])
+    ax.coastlines('50m', edgecolor='black', linewidth=0.5)
+    ax.add_feature(states_provinces, edgecolor='black', linewidth=0.5)
+    ax.add_feature(country_borders, edgecolor='black', linewidth=0.5)
+    return ax
+
+
+###########################################
+
+# Open the example netCDF data
+ds = netCDF4.Dataset(get_test_data('gfs_output.nc', False))
+print(ds)
+
+###########################################
+
+# Convert number of hours since the reference time into an actual date
+time_vals = netCDF4.num2date(ds.variables['time'][:].squeeze(), ds.variables['time'].units)
+
+###########################################
+
+# Combine 1D latitude and longitudes into a 2D grid of locations
+lon_2d, lat_2d = np.meshgrid(ds.variables['lon'][:], ds.variables['lat'][:])
+
+###########################################
+
+# Assign units
+vort_500 = ds.variables['vort_500'][0] * units('1/s')  # To plot a e-5 s^-1
+surface_temp = ds.variables['temp'][0] * units.kelvin
+precip_water = ds.variables['precip_water'][0] * units.mm
+winds_300 = ds.variables['winds_300'][0] * units('m/s')
+
+###########################################
+
+# Do unit conversions to what we wish to plot
+vort_500 = vort_500 * 1e5
+surface_temp = surface_temp.to('degF')
+precip_water = precip_water.to('inches')
+winds_300 = winds_300.to('knots')
+
+###########################################
+
+# Smooth the height data
+heights_300 = ndimage.gaussian_filter(ds.variables['heights_300'][0], sigma=1.5, order=0)
+heights_500 = ndimage.gaussian_filter(ds.variables['heights_500'][0], sigma=1.5, order=0)
+
+###########################################
+
+# Add state boundaries to plot
+states_provinces = cfeature.NaturalEarthFeature(category='cultural',
```

Comment:
Not wild about using these as global variables, but not sure of a better way to do it that's new-user-accessible.

How about at least moving these up to right before the `plot_background` function where they're used?